### ISSUE TITLE
pygame.draw.ellipse does not draw ellipses with odd sizes

### ISSUE THREAD

**Issue by illume** [pygame/pygame#233]
Steps to fix this issue of "pygame.draw.ellipse not being able to draw an odd sized ellipse".

- [x] ~~write a unit test that shows the problem of odd sizes not being possible inside [tests/draw_test.py](https://github.com/pygame/pygame/blob/master/tests/draw_test.py)~~ (**good first issue** DONE)
- [x] fix draw ellipse in [src_c/draw.c](https://github.com/pygame/pygame/blob/master/src_c/draw.c) (harder, not a good first issue)

https://www.pygame.org/docs/ref/draw.html#pygame.draw.ellipse

---

Originally reported by: **donLorenzo (Bitbucket: [donLorenzo](https://bitbucket.org/donLorenzo), GitHub: [donLorenzo](https://github.com/donLorenzo))**

----------------------------------------

Reported by Florian Krause on the [pygame mailing list](http://article.gmane.org/gmane.comp.python.pygame/25539):

```
#!plain

I was actually wrong. It seems even worse. Pygama is not capable to draw ellipses with odd sizes at all. It will always be the same ellipse as x-1, y-1.
E.g. an elipse with size 3,3 will result in the same ellipse as the one with size 2,2.
Can anyone reproduce this?
```

Code to reproduce:

```
#!python

import pygame
pygame.init()
screen = pygame.display.set_mode((100, 100))
pygame.draw.ellipse(screen, (255,0,0), (0,0,5,5))
pygame.display.update()
```
The resulting ellipse will have a 4x4 size

----------------------------------------
- Bitbucket: https://bitbucket.org/pygame/pygame/issue/233

**Comment by Earlo** [comment]
Did #511 fix this? Should this be closed?

**Comment by charlesej** [comment]
Yes and yes. This issue has been resolved by PR #511.
